libs/core/src/lib/utils/directives/resize/resize.directive.ts

```diff
@@ -0,0 +1,205 @@
+
+    /** @hidden */
+    private _subscriptions = new Subscription();
+
+    constructor(private _elementRef: ElementRef) {
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes['disabled']) {
+            if (changes['disabled'].previousValue === false && changes['disabled'].currentValue === true) {
+                this._subscriptions.unsubscribe();
+            } else if (changes['disabled'].previousValue === true && changes['disabled'].currentValue === false) {
+                this._setResizeListeners();
+            }
+        }
+    }
+
+    /** @hidden */
+    ngAfterContentInit() {
+        if (!this.disabled) {
+            this._setResizeListeners();
+        }
+    }
+
+    /** @hidden */
+    ngOnDestroy() {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden Sets Resize listeners */
+    private _setResizeListeners(): void {
+
+        const resize = this._getResizeFunction();
+        const moveOffset = this._getMoveOffsetFunction();
+        const resizeContainer = this._findResizeContainer();
+        const isBoundaryOverflow = this._getBoundaryOverflowFunction(resizeContainer);
+
+        const mouseUpEvent$ = fromEvent(window, 'mouseup');
+        const mouseMoveEvent$ = fromEvent(resizeContainer, 'mousemove');
+        const mouseDownEvent$ = fromEvent(this.resizeHandleReference.elementRef.nativeElement, 'mousedown');
+
+        const resizeActive$ = merge(mouseDownEvent$.pipe(mapTo(true)), mouseUpEvent$.pipe(mapTo(false)));
+        const emitResizableEvents$ = this._getResizeEventsNotifiers(resizeActive$);
+        const preventOtherPointerEvents$ = this._blockOtherPointerEvents(resizeActive$);
+
+        const resizingCursorPosition$ = mouseMoveEvent$.pipe(
+            withLatestFrom(resizeActive$),
+            filter(([_, isActive]: [MouseEvent, boolean]) => isActive),
+            map(([event, _]) => moveOffset(event)),
+            filter(move => isBoundaryOverflow(move))
+        );
+
+        this._subscriptions.add(resizingCursorPosition$.subscribe(event => resize(event)));
+        this._subscriptions.add(preventOtherPointerEvents$.subscribe());
+        this._subscriptions.add(emitResizableEvents$.subscribe());
+    }
+
+    /** @hidden Creates resize function*/
+    private _getResizeFunction(): (move: ResizeMove) => void {
+        return (move: ResizeMove) => {
+            this._elementRef.nativeElement.style.width = this._elementRef.nativeElement.offsetWidth + move.x + 'px';
+            this._elementRef.nativeElement.style.height = this._elementRef.nativeElement.offsetHeight + move.y + 'px';
+        }
+    }
+
+    /** @hidden Creates move function */
+    private _getMoveOffsetFunction(): (event: MouseEvent) => ResizeMove {
+
+        let verticalModifier: 1 | -1;
+        let horizontalModifier: 1 | -1;
+
+        switch (this.resizeHandleLocation) {
+            case 'top-left':
+                horizontalModifier = -1;
```

Comment:
you can define variable instead typing everywhere `1` and `-1`. Its a good practice because value might change